src/lib/raw_data/xml_reader.cpp

```diff
@@ -556,6 +562,13 @@ std::optional<RawData::RawData> XmlReader::read_mzml(
                             if (accession == "MS:1000521") {
                                 precision = 32;
                             }
+                            // Compression.
+                            if (accession == "MS:1000576") {
+                                compressed = 0;
+                            }
+                            if (accession == "MS:1000574") {
```

Comment:
these things are mutually exclusive, it shouldn't be a problem though. In any case, using the "no compression" accession seems unnecessary, since we only have 2 cases (zlib compressed or not compressed). Make the not compressed the default case and only change it if `MS:1000574` is present.

I know you looked at the compressed/not compressed files, but in case you need it this is the page with all extensions:
https://raw.githubusercontent.com/HUPO-PSI/psi-ms-CV/master/psi-ms.obo

And here are the specs of the format:

https://www.psidev.info/mzML

Even though there may be other compressions in mzML files, for now I think it's safe to only consider zlib compressed or not compressed spectra.